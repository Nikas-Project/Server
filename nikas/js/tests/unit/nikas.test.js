"use strict";

test('Create Postbox', () => {
    // Set up our document body
    document.body.innerHTML =
        '<div id=nikas-thread></div>' +
        // Note: `src` and `data-nikas` need to be set,
        // else `api` fails to initialize!
        '<script src="http://nikas.api/js/embed.min.js" data-nikas="/"></script>';

    const nikas = require("app/nikas");
    const $ = require("app/dom");

    const config = require("app/config");
    const i18n = require("app/i18n");
    const svg = require("app/svg");

    const template = require("app/template");

    template.set("conf", config);
    template.set("i18n", i18n.translate);
    template.set("pluralize", i18n.pluralize);
    template.set("svg", svg);

    var nikas_thread = $('#nikas-thread');
    nikas_thread.append('<div id="nikas-root"></div>');
    nikas_thread.append(new nikas.Postbox(null));

    // Will create a `.snap` file in `./__snapshots__/`.
    // Don't forget to check in those files when changing anything!
    expect(nikas_thread.innerHTML).toMatchSnapshot();
});
