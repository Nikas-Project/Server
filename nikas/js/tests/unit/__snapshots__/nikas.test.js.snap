// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Postbox 1`] = `"<div id=\\"nikas-root\\"></div><div class=\\"nikas-postbox\\"><div class=\\"nikas-form-wrapper\\"><div class=\\"nikas-textarea-wrapper\\"><div class=\\"nikas-textarea nikas-placeholder\\" contenteditable=\\"true\\">Type Comment Here (at least 3 chars)</div><div class=\\"nikas-preview\\"><div class=\\"nikas-comment\\"><div class=\\"nikas-text-wrapper\\"><div class=\\"nikas-text\\"></div></div></div></div></div><section class=\\"nikas-auth-section\\"><p class=\\"nikas-input-wrapper\\"><input type=\\"text\\" name=\\"author\\" placeholder=\\"Name (optional)\\" value=\\"\\"></p><p class=\\"nikas-input-wrapper\\"><input type=\\"email\\" name=\\"email\\" placeholder=\\"E-mail (optional)\\" value=\\"\\"></p><p class=\\"nikas-input-wrapper\\"><input type=\\"text\\" name=\\"website\\" placeholder=\\"Website (optional)\\" value=\\"\\"></p><p class=\\"nikas-post-action\\"><input type=\\"submit\\" value=\\"Submit\\"></p><p class=\\"nikas-post-action\\"><input type=\\"button\\" name=\\"preview\\" value=\\"Preview\\"></p><p class=\\"nikas-post-action\\"><input type=\\"button\\" name=\\"edit\\" value=\\"Edit\\"></p></section><section class=\\"nikas-notification-section\\" style=\\"display: none;\\"><label><input type=\\"checkbox\\" name=\\"notification\\">Subscribe to email notification of replies</label></section></div></div>"`;
